pandas/core/generic.py

```diff
@@ -3630,6 +3630,85 @@ def interpolate(self, method='linear', axis=0, limit=None, inplace=False,
             return res
 
     # ----------------------------------------------------------------------
+    # Timeseries methods Methods
+
+    def asof(self, where, subset=None):
+        """
+        Return last good (non-NaN) value in object if subset is NaN for
+        requested date.
```

Comment:
I think it should be clarified that for dataframe, the last row without any NaN is taken (or the last row without NaN considering only the subset of columns)